lib/ImportFromC/HeaderToModel.cpp

```diff
@@ -639,43 +672,62 @@ bool DeclVisitor::VisitFunctionDecl(const clang::FunctionDecl *FD) {
                                                   Errors);
       if (not Register.has_value()) {
         if (not Stack.has_value()) {
-          Errors.emplace_back("import-from-c: Arguments must have either a "
-                              "register or a stack annotation.\n");
+          Errors.emplace_back("import-from-c failed: Argument #"
+                              + std::to_string(I)
+                              + " is missing it's location annotation.\n");
+          Errors.emplace_back("                      Please add either "
+                              "`_REG($name)` or `_STACK`.\n");
           return false;
         } else {
           if (I != N - 1) {
-            Errors.emplace_back("import-from-c: Only the very last RFT "
+            Errors.emplace_back("import-from-c failed: Only the very last RFT "
                                 "argument is allowed to represent stack, which "
                                 "also means there can only be one.\n");
+            Errors.emplace_back("                      Please either remove "
+                                "`_STACK` annotation from the argument #"
+                                + std::to_string(I)
+                                + " or move it into the stack argument "
+                                  "struct.\n");
             return false;
           }
 
           if (not ParamType->isStruct()) {
-            Errors.emplace_back("import-from-c: RFT stack argument must be a "
-                                "struct.\n");
+            // TODO: should we allow non-structs here? The only downside is that
+            //       it would mean transparently adding an extra wrapping struct
+            //       to the model, which doesn't sounds too bad to me.
```

Comment:
In principle we could, but for now there's no advantage, and lots of places in the code still assume it's a struct.

Reply:
No, that's not what I meant!

I meant allowing the users to specify a non-struct, and then we transparently make a new struct for them to wrap whatever type they gave.

Reply:
no, I'd like to minimize surprise for users. and a magical new wrapper struct is surprising. let's leave it like it is for now

Reply:
Gone!